docs/schema-definition.md

```diff
@@ -116,60 +114,64 @@ In this case, it is recommended to pass **typeLoader** option to schema construc
 of your object **fields** as callbacks.
 
 Type loading concept is very similar to PHP class loading, but keep in mind that **typeLoader** must
-always return the same instance of a type.
+always return the same instance of a type. A good way to ensure this is to use a type registry:
 
-Usage example:
 ```php
-<?php
+<?php declare(strict_types=1);
+use GraphQL\Type\Definition\Type;
 use GraphQL\Type\Definition\ObjectType;
 use GraphQL\Type\Schema;
 
-class Types
+class TypeRegistry
 {
-    private $types = [];
+    /**
+     * @var array<string, Type> 
+     */
+    private array $types = [];
 
-    public function get($name)
+    public function __invoke(string $name): Type
     {
-        if (!isset($this->types[$name])) {
-            $this->types[$name] = $this->{$name}();
-        }
-        return $this->types[$name];
+        return $this->types[$name]
+            ??= $this->{$name}();
     }
 
-    private function MyTypeA()
+    private function MyTypeA(): ObjectType
     {
         return new ObjectType([
             'name' => 'MyTypeA',
-            'fields' => function() {
-                return [
-                    'b' => ['type' => $this->get('MyTypeB')]
-                ];
-            }
+            'fields' => fn() => [
+                'b' => [
+                    'type' => $this('MyTypeB')
+                ],
+            ]
         ]);
     }
-    
-    private function MyTypeB()
+
+    private function MyTypeB(): ObjectType
     {
         // ...
     }
 }
 
-$registry = new Types();
+$typeRegistry = new TypeRegistry();
 
 $schema = new Schema([
-    'query' => $registry->get('Query'),
-    'typeLoader' => function($name) use ($registry) {
-        return $registry->get($name);
-    }
+    'query' => $typeRegistry('Query'),
```

Comment:
TBH I liked `get` much better. This reads like `type registry query WHAT?`. IMO too fancy which is not necessary.

Reply:
The line below becomes much nicer though and shaves off a level of indirection.

Reply:
It is objectively worse in terms of performance though.

Reply:
I guess it adds micro overhead. But for docs it should rather be verbose.

Reply:
Fine.